### ISSUE TITLE
Ability to tag a different instance so toot will appear in its local timeline

### ISSUE THREAD

**Issue by craigS89** [mastodon/mastodon#6865]
**Feature request**: Using a tagging symbol, for example '+', a user can tag a federated instance so the toot will appear in that instance's local timeline.

**Example**: "Hey, can anyone on the +linux.social instance recommend a good distro for a newbie to try?"

**Benefits** This feature would benefit instances that are dedicated to topics and used like dedicated forums and let users gain greater exposure to other users with these interests across instances.
This would also allow for more passive inclusion into topics and discovery than hashtags, which require actively searching the specific terms the user may not yet be aware of.
This may also help new users, who find the idea of choosing an instance daunting. knowing they can interact directly with an instance's users even if they do not belong to it may help make choosing an instance to feel less limiting.

**Security and privacy**: this may carry a risk of being a vector for types of harassment, so I believe admins should be able to turn off this feature and for tagging to only be available between federated instances. 

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [ ] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by theboss** [comment]
I implemented a feature similar to this request in my Mastodon instance `theboss.tech`. I customized Mastodon and replaced the local TL with tag TL.

The local TL of my instance is now the tag TL `#theboss_tech`. Therefore, if a user post this tag on other instance, it appears in the TL of my instance. (the tag can be changed in `.env.production`)

Also I implemented default-tagging feature. The default-tag `#theboss_tech` is automatically added to posts when users post with public.

When a user wants to post only to another tag TL, if the user manually tags and clicks the new button "Toot without Local", the post does not appear in the default-tag TL and be posted to any tag TL. (If the user clicks this button without tag, the visibility will be changed to unlisted.)

Changes to the source code are included in [this branch](https://github.com/theboss/mastodon/tree/feature-federated-local-timeline). (requires configure `.env.production`)

I have been running the customized instance since February, it works well.

I will not PR this feature because I consider this feature may be contrary to Mastodon's philosophy. This feature does not work on a new instance, as it requires that the instance is already federated with many users. In addition, this may make the user lose motivation of remote following.
